muspinsim/input/input.py

```diff
@@ -117,31 +118,48 @@ def __init__(self, fs=None):
                 pass
 
             # Now parse
+            errors_found = []
             for header, block in raw_blocks.items():
+                try:
+                    hsplit = header.split()
+                    name = hsplit[0]
+                    args = hsplit[1:]
 
-                hsplit = header.split()
-                name = hsplit[0]
-                args = hsplit[1:]
+                    try:
+                        KWClass = InputKeywords[name]
+                    except KeyError:
+                        raise MuSpinInputError(
+                            "Invalid keyword {0} found in input file".format(name)
+                        )
 
-                try:
-                    KWClass = InputKeywords[name]
-                except KeyError:
-                    raise MuSpinInputError(
-                        "Invalid keyword " "{0} ".format(name) + "found in input file"
-                    )
+                    if issubclass(KWClass, MuSpinEvaluateKeyword):
+                        kw = KWClass(block, args=args, variables=self._variables)
+                    else:
+                        kw = KWClass(block, args=args)
 
-                if issubclass(KWClass, MuSpinEvaluateKeyword):
-                    kw = KWClass(block, args=args, variables=self._variables)
-                else:
-                    kw = KWClass(block, args=args)
+                    kwid = kw.id
 
-                kwid = kw.id
+                    if name != kwid:
+                        self._keywords[name] = self._keywords.get(name, {})
+                        self._keywords[name][kwid] = kw
+                    else:
+                        self._keywords[name] = kw
+                except ValueError as e:
+                    errors_found += [e]
+
+            if errors_found:
+                tbs = ""
+                for i, e in enumerate(errors_found):
+                    tbs += "Error {0}\n{1}\n".format(
+                        i + 1,
+                        "".join(traceback.format_exception(None, e, e.__traceback__)),
+                    )
 
-                if name != kwid:
-                    self._keywords[name] = self._keywords.get(name, {})
-                    self._keywords[name][kwid] = kw
-                else:
-                    self._keywords[name] = kw
+                raise MuSpinInputError(
+                    "Errors found whilst trying to parse input file: \n\n{0}".format(
+                        tbs
+                    )
+                )
```

Comment:
With the other stack traces removed this is no longer a big problem in the CLI. However, having watched how Leandro used the Galaxy UI today, I think it might be preferable to print the errors to console and use `sys.exit` or something instead, rather than raising an error with a stack trace that fills up the whole dataset info box. The Galaxy user might still not be able to see all the errors, but they can at least get an immediate idea of what went wrong if the first thing they see is "N errors found while parsing..."